### PR TITLE
cnojima's requested changes to the git RPCs

### PR DESCRIPTION
**Pull request by oconnor663**
- Include the last modifying device name.
- Synthesize the keybase:// repo URL to keep that logic out of the GUI.
- Synthesize a globally unique repo ID also.

r? @mlsteele

### CODE REVIEW

File: go/git/put_get_test.go [hunk 1]
```diff
@@ -86,6 +88,9 @@ func TestPutAndGet(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, 1, len(oneRepo), "expected to get only one repo back, found: %d", len(oneRepo))
 	require.Equal(t, "repoNameFirst", string(oneRepo[0].LocalMetadata.RepoName))
+	require.Equal(t, kbtest.DefaultDeviceName, oneRepo[0].ServerMetadata.LastModifyingDeviceName)
+	require.Equal(t, string(team1.Chain.Id+"_abc123"), oneRepo[0].GlobalUniqueID)
+	require.Equal(t, "keybase://team/"+teamName1+"/repoNameFirst", oneRepo[0].RepoUrl)
```
Comment (mlsteele) [inline go/git/put_get_test.go]: this doesn't make sense for implicit teams, is that a problem?

Comment (mlsteele) [reply go/git/put_get_test.go]: nevermind, it's fine

Comment (oconnor663) [reply go/git/put_get_test.go]: This is generated from the repo name and the `Folder`, so for an implicit team we'd expect either `keybase://private/...` or `keybase://public/...`. Does that sound right?

**Comment by oconnor663** [comment]
@chrisnojima could you eyeball the protocol changes here and see if they look right to you?

File: go/git/put_get_test.go (hunk 1 above)
Comment (mlsteele) [reply go/git/put_get_test.go]: yes. Might be worth adding a `GetMetadata` to `TestPutAndGetImplicitTeam` below

File: go/git/get.go [hunk 2]
```diff
@@ -178,15 +187,31 @@ func getMetadataInner(ctx context.Context, g *libkb.GlobalContext, folder *keyba
...
+		upak, _, err := g.GetUPAKLoader().LoadV2(libkb.LoadUserArg{Name: responseRepo.LastModifyingUsername})
+		var deviceName string
+		for _, deviceKey := range upak.Current.DeviceKeys {
+			if deviceKey.DeviceID.Eq(responseRepo.LastModifyingDeviceID) {
+				deviceName = deviceKey.DeviceDescription
+				break
+			}
+		}
+		if deviceName == "" {
+			return nil, fmt.Errorf("can't find device name for %s's device ID %s", upak.Current.Username, responseRepo.LastModifyingDeviceID)
```
Comment (mlsteele) [inline go/git/get.go]: Do they want this to be fatal? Maybe they'd prefer empty, idk. Maybe it will never happen.

**Comment by chrisnojima** [comment]
seems fine. since your doing this want to add the deleteRPC

File: go/git/get.go (hunk 2 above)
Comment (oconnor663) [reply go/git/get.go]: As far as I can tell this should never happen, and so I want us to have to think about it if it turns out it does :)

**Comment by keybase-ci-visdiff** [comment]
The commits b32c51a8ca7f...4a0a7b760a87 introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Markdown-emoji** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1505496991911-b32c51a8ca7f-4a0a7b760a87-linux/Markdown-emoji-b32c51a8ca7f.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1505496991911-b32c51a8ca7f-4a0a7b760a87-linux/Markdown-emoji-4a0a7b760a87.png)  
   Markdown-emoji comparison
</summary>

File: go/git/put_get_test.go (hunk 1 above)
Comment (oconnor663) [reply go/git/put_get_test.go]: good call, adding this caught a bug in some old code that was assuming the `TEAM` type.

**Pull request merged**